wayback/_client.py

```diff
@@ -609,6 +609,250 @@ def search(self, url, *, match_type=None, limit=1000, offset=None,
+            If true, don't yield records that look like they have no actual
+            memento associated with them. Some crawlers will erroneously
+            attempt to capture bad URLs like
+            ``http://mailto:someone@domain.com`` or
+            ``http://data:image/jpeg;base64,AF34...`` and so on. This is a
+            filter performed client side and is not a CDX API argument.
+        page_size : int, default: 5
+            How many *index blocks* (not results!) to search in each request.
+            Regardless of the value set here, this function will yield results
+            from every page. If you want to stop early, then stop iterating
+            over the results of this method.
+
+        Raises
+        ------
+        UnexpectedResponseFormat
+            If the CDX response was not parseable.
+
+        Yields
+        ------
+        version: CdxRecord
+            A :class:`CdxRecord` encapsulating one capture or revisit
+
+        References
+        ----------
+        * HTTP API Docs: https://github.com/internetarchive/wayback/tree/master/wayback-cdx-server
+        * SURT formatting: http://crawler.archive.org/articles/user_manual/glossary.html#surt
+        * SURT implementation: https://github.com/internetarchive/surt
+
+        Notes
+        -----
+        Several CDX API parameters are not relevant or handled automatically
+        by this function. This does not support: `page`, `offset`, `limit`,
+        `output`, `fl`, `showDupeCount`, `showSkipCount`, `lastSkipTimestamp`,
+        `showNumPages`, `showPagedIndex`.
+        """
+        if (not isinstance(page_size, int)) or page_size <= 0:
+            raise TypeError(f'`page_size` must be a positive integer, not {page_size!r}')
+
+        # TODO: support args that can be set multiple times: filter, collapse
+        # Should take input as a sequence and convert to repeat query args
+        # TODO: Check types
+        query_args = {'url': url, 'matchType': match_type, 'from': from_date,
+                      'to': to_date, 'filter': filter_field,
+                      'collapse': collapse, 'resolveRevisits': resolve_revisits,
+                      'pageSize': page_size}
+
+        query = {}
+        for key, value in query_args.items():
+            if value is not None:
+                if isinstance(value, str):
+                    query[key] = value
+                elif isinstance(value, date):
+                    query[key] = _utils.format_timestamp(value)
+                else:
+                    query[key] = str(value).lower()
+
+        # Since pages are a number of *blocks searched* and not results, a page
+        # in the middle of the result set may have nothing in it. The only way
+        # to know when to stop iterating is to check how many pages there are.
+        page_count = int(self.session.request('GET', CDX_SEARCH_2_URL, params={
+            **query,
+            'showNumPages': 'true'
+        }).text)
```

Comment:
This might not be necessary! I just discovered that sending too high a `page` value gets a 400 error with the header `x-archive-wayback-runtime-error: page must be smaller than numpages`, so we can in theory check for that and stop.

That said, that’s a very human-readable message and feels unstable. We should check with folks at the Internet Archive about what approach they’d prefer people use.